graph-db/serverless/src/utils/cmr/deleteCmrCollection.js

```diff
@@ -8,22 +8,45 @@ const { P: { lte } } = gremlin.process
  * Delete the collection with the given concept id from graph db
  * @param {String} conceptId Collection concept id from CMR
  * @param {Gremlin Traversal Object} gremlinConnection connection to gremlin server
+ * @param {String} vertexLabel Label of the linked vertex
+ * @param {String} edgeName Name of the edge between the dataset vertex and its linked vertex
  * @returns
  */
-export const deleteCmrCollection = async (conceptId, gremlinConnection) => {
-  // drop all the documentation vertices that are connected to and only connected to the dataset vertex
+const deleteLinkedVertices = async (conceptId, gremlinConnection, vertexLabel, edgeName) => {
   try {
     await gremlinConnection
       .V()
       .has('dataset', 'concept-id', conceptId)
-      .outE('documentedBy')
+      .outE(edgeName)
       .inV()
-      .where(gremlinStatistics.inE('documentedBy').count().is(lte(1)))
+      .where(gremlinStatistics.inE(edgeName).count().is(lte(1)))
       .drop()
       .next()
   } catch (error) {
-    console.error(`Error deleting documentation vertices only linked to collection [${conceptId}]: ${error.message}`)
+    console.error(`Error deleting ${vertexLabel} vertices only linked to collection [${conceptId}]: ${error.message}`)
+
+    return false
+  }
+
+  return true
+}
+/**
+ * Delete the collection with the given concept id from graph db
+ * @param {String} conceptId Collection concept id from CMR
+ * @param {Gremlin Traversal Object} gremlinConnection connection to gremlin server
+ * @returns
+ */
+export const deleteCmrCollection = async (conceptId, gremlinConnection) => {
+  // drop all the campaign vertices that are connected to and only connected to the dataset vertex
+  let success
+  success = await deleteLinkedVertices(conceptId, gremlinConnection, 'campaign', 'includedIn')
+  if (success === false) {
+    return false
+  }
 
+  // drop all the documentation vertices that are connected to and only connected to the dataset vertex
+  success = await deleteLinkedVertices(conceptId, gremlinConnection, 'documentation', 'documentedBy')
+  if (success === false) {
```

Comment:
Can the early outs be simplified to `if (!success)`

Reply:
`!success` will be `true` if `success` is undefined or `false`. I may not matter in this particular case, but I think it is better to check for `false` specifically as that is what is returned from the function.